From a1fbae64456ac3e82e9ffea8e4394df23b2e148c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ji=C5=99=C3=AD=20Klime=C5=A1?= <jklimes@redhat.com>
Date: Tue, 22 Oct 2013 15:01:06 +0200
Subject: [PATCH 1/1] ifcfg-rh: do not crash (assertion) when writing bonding
 slaves (rh #1021947)

Brigde or bonding slaves do not contain IP{4|6} setting in the connection.

https://bugzilla.redhat.com/show_bug.cgi?id=1021947
---
 system-settings/plugins/ifcfg-rh/writer.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/system-settings/plugins/ifcfg-rh/writer.c b/system-settings/plugins/ifcfg-rh/writer.c
index d16078d..6825296 100644
--- a/system-settings/plugins/ifcfg-rh/writer.c
+++ b/system-settings/plugins/ifcfg-rh/writer.c
@@ -1658,21 +1658,24 @@ out:
 
 static void
 write_ip4_aliases (NMConnection *connection, char *base_ifcfg_path)
 {
 	NMSettingIP4Config *s_ip4;
 	char *base_ifcfg_dir, *base_ifcfg_name;
 	int i, num, base_ifcfg_path_len, base_ifcfg_name_len;
 	GDir *dir;
 
 	s_ip4 = (NMSettingIP4Config *) nm_connection_get_setting (connection, NM_TYPE_SETTING_IP4_CONFIG);
-	g_assert (s_ip4 != NULL);
+
+	/* Bridge/bond slaves don't contain IP setting */
+	if (!s_ip4)
+		return;
 
 	base_ifcfg_path_len = strlen (base_ifcfg_path);
 	base_ifcfg_dir = g_path_get_dirname (base_ifcfg_path);
 	base_ifcfg_name = g_path_get_basename (base_ifcfg_path);
 	base_ifcfg_name_len = strlen (base_ifcfg_name);
 
 	/* Remove all existing aliases for this file first */
 	dir = g_dir_open (base_ifcfg_dir, 0, NULL);
 	if (dir) {
 		const char *item;
-- 
1.7.11.7

